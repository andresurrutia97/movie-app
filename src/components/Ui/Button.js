import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ children, onClick, type }) => {
  const baseClass = 'w-full focus:outline-none text-white py-2 px-4 rounded-md';

  let classes;

  switch (type) {
    case 'outlined':
      classes = [baseClass, 'border border-2 custom-blue-border bg-black'].join(
        ' '
      );
      break;

    default:
      classes = [baseClass, 'custom-blue-bg'].join(' ');
  }

  return (
    <button type="button" onClick={onClick} className={classes}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.string,
  onClick: PropTypes.func,
  type: PropTypes.string,
};

Button.defaultProps = {
  children: 'Click me',
  onClick: () => {},
  type: '',
};

export default Button;
