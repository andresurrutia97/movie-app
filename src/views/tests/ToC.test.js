import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import { useAuth0 } from '@auth0/auth0-react';

import TOCView from '../ToC';
import Button from '../../components/Ui/Button';

jest.mock('@auth0/auth0-react');

describe('<TOC/>', () => {
  const component = (
    <BrowserRouter>
      <TOCView />
    </BrowserRouter>
  );
  const loginMock = jest.fn();
  beforeEach(() => {
    useAuth0.mockReturnValue({
      loginWithRedirect: loginMock,
      isAuthenticated: false,
    });
  });

  it('should render correcty ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should not display a login Button if the user is authenticated', () => {
    useAuth0.mockReturnValue({
      isAuthenticated: true,
    });
    const wrapper = mount(component);
    expect(wrapper.find(Button)).toHaveLength(0);
  });

  it('should display a login Button if the user is not authenticated', () => {
    useAuth0.mockReturnValue({
      isAuthenticated: false,
    });
    const wrapper = mount(component);
    expect(wrapper.find(Button)).toHaveLength(1);
  });

  it('should call the login function when click on the login button', () => {
    const wrapper = mount(component);

    wrapper.find(Button).simulate('click');
    expect(loginMock).toBeCalledTimes(1);
  });
});
