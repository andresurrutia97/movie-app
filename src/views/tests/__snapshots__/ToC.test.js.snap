// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TOC/> should render correcty  1`] = `
<div
  className="pt-16 my-4 mx-5 md:w-6/12 md:mx-auto md:mt-16"
>
  <div
    className="flex flex-col items-center md:flex-row md:items-start md:justify-between"
  >
    <div
      className="text-2xl text-white md:text-6xl text-center md:text-left"
    >
      TOC & Privacy Policy 
    </div>
    <div
      className="w-6/12 my-5 md:w-1/4 md:mt-8"
    >
      <button
        className="w-full focus:outline-none text-white py-2 px-4 rounded-md custom-blue-bg"
        onClick={[Function]}
        type="button"
      >
        LOGIN
      </button>
    </div>
  </div>
  <hr
    className="mb-5 border-gray-600"
  />
  <div
    className="mb-5 text-justify"
  >
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam urna tellus, tincidunt sit amet vestibulum sed, semper sed leo. Integer eget est augue. Sed consectetur tempor maximus. Praesent sapien nisl, aliquet eget aliquam sed, convallis auctor ex. Nulla tristique quis mi quis iaculis. In facilisis purus a tempor sollicitudin. Duis eu ipsum et ipsum rutrum dapibus vitae quis ex. Donec id nisi eget nulla facilisis sodales. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Aenean condimentum justo quis risus posuere, non laoreet ante iaculis. Etiam lobortis ullamcorper tristique. Vivamus suscipit turpis quis nunc tincidunt, eget placerat nibh aliquam. Vivamus sodales at purus sed euismod. Sed a feugiat orci. Integer ligula libero, facilisis in erat quis, pellentesque bibendum massa. In vestibulum ante in eleifend varius. Ut id erat sit amet nunc imperdiet venenatis ut ut lorem. Duis dictum viverra diam sit amet eleifend. Mauris lacus nulla, maximus ac placerat sit amet, viverra non nulla. Ut malesuada magna ut enim bibendum maximus. Etiam ac elementum nibh, sit amet molestie tortor. Vestibulum quis iaculis leo. Donec ante elit, tristique congue erat et, porta finibus lacus. Pellentesque vestibulum id eros non fermentum. Nulla id hendrerit ex. Nullam fringilla mauris ut ligula scelerisque luctus. Ut rhoncus sed tortor a vulputate. Vestibulum vel odio at elit posuere commodo porta eget tortor. Vestibulum eget lectus nunc. Sed venenatis viverra ligula at vestibulum. Sed nulla metus, efficitur in nisi ac, efficitur sodales massa.
  </div>
  <hr
    className="mb-5 border-gray-600"
  />
</div>
`;
